lib/github-pages-health-check/domain.rb

```diff
@@ -388,9 +388,19 @@ def enforces_https?
 
       # Can an HTTPS certificate be issued for this domain?
       def https_eligible?
-        (cname_to_github_user_domain? || pointed_to_github_pages_ip?) &&
-          !aaaa_record_present? && !non_github_pages_ip_present? &&
-          caa.lets_encrypt_allowed?
+        # Can't have any IP's which aren't GitHub's present.
+        return false if non_github_pages_ip_present?
+        # Can't have any AAAA records present
+        return false if aaaa_record_present?
+        # Must be a CNAME or point to our IPs.
+
+        # Only check the one domain if a CNAME. Don't check the parent domain.
+        return caa.lets_encrypt_allowed? if cname_to_github_user_domain?
```

Comment:
```suggestion
        return cname_to_github_user_domain?
```

I don't think we even need to check the `CAA` record for the custom domain when it is a `CNAME`, as a `CNAME` can't co-exist with another record type.

This is why for example you can't use an `CNAME` at the apex of a domain, as it'd be co-existing with `NS` records at a minimum.

In this case, we only need to worry about `CAA` records when the custom domain is using `A` record(s), and can completely ignore `CAA` records when it is a `CNAME` once we have `CAA` records in place for `*.github.io`.

Ref: Section 3.6.2 in https://tools.ietf.org/html/rfc1034